root/usr/sbin/ns8-join

```diff
@@ -100,38 +102,31 @@ api_endpoint = f'https://{args.host}/cluster-admin'
 # POST login request
 req = request.Request(f'{api_endpoint}/api/login', data=data)
 req.add_header('Content-Type', 'application/json')
-resp = request.urlopen(req, context=ctx)
+
+try:
+    resp = request.urlopen(req, context=ctx)
+except error.URLError as ex:
+    print("ns8-join:", str(ex), file=sys.stderr)
+    sys.exit(1)
 
 payload = json.loads(resp.read())
 
 # Retrieve account provider information
 account_provider_json = subprocess.check_output(['/usr/sbin/account-provider-test', 'dump'], encoding='utf-8')
 account_provider_config = json.loads(account_provider_json)
 
-# Prepare arguments for add-node
-priv_key = subprocess.check_output(['cat', '/var/lib/nethserver/secrets/ns8wg']).decode('ASCII').rstrip()
-pub_key = subprocess.check_output(['wg', 'pubkey'], input=priv_key.encode('ASCII')).decode('ASCII').rstrip()
+# Generate a new Wireguard key. If add-node succeedes, but VPN connection
+# fails, we must not reuse the same key to avoid conflicts.
+tmpmask = os.umask(0o077)
+with open('/var/lib/nethserver/secrets/ns8wg', 'w') as wgfp:
+    priv_key = subprocess.Popen(['wg', 'genkey'], stdout=wgfp)
+with open('/var/lib/nethserver/secrets/ns8wg', 'r') as wgfp:
+    pub_key = subprocess.check_output(['wg', 'pubkey'], stdin=wgfp, encoding='UTF-8').rstrip("\n")
+os.umask(tmpmask) ; del tmpmask
```

Comment:
Why not keep the check_output call? It raises an exception in case of error, IMO it's better.